README.tfdoc.hcl

```diff
@@ -0,0 +1,780 @@
+    content = <<-END
+      See [variables.tf] and [examples/] for details and use-cases.
+    END
+
+    section {
+      title = "Module Configuration"
+
+      variable "module_enabled" {
+        type        = bool
+        default     = true
+        description = <<-END
+          Specifies whether resources in the module will be created.
+        END
+      }
+
+      variable "module_depends_on" {
+        type           = list(dependency)
+        description    = <<-END
+          A list of dependencies. Any object can be _assigned_ to this list to define a hidden external dependency.
+        END
+        readme_example = <<-END
+          module_depends_on = [
+            google_network.network
+          ]
+        END
+      }
+    }
+
+    section {
+      title = "Main Resource Configuration"
+
+      variable "name" {
+        required    = true
+        type        = string
+        description = <<-END
+          Name of the bucket.
+        END
+      }
+
+      variable "force_destroy" {
+        type        = bool
+        default     = false
+        description = <<-END
+          When deleting a bucket, this boolean option will delete all contained objects. If you try to delete a bucket that contains objects, Terraform will fail that run.
+        END
+      }
+
+      variable "location" {
+        type        = string
+        default     = "US"
+        description = <<-END
+          The GCS location.
+        END
+      }
+
+      variable "project" {
+        type        = string
+        description = <<-END
+          The ID of the project in which the resource belongs. If it is not provided, the provider project is used.
+        END
+      }
+
+      variable "storage_class" {
+        type        = string
+        default     = "STANDARD"
+        description = <<-END
+          The Storage Class of the new bucket. Supported values include: `STANDARD`, `MULTI_REGIONAL`, `REGIONAL`, `NEARLINE`, `COLDLINE`, `ARCHIVE`.
+        END
+      }
+
+      variable "lifecycle_rules" {
+        type           = list(lifecycle_rule)
+        description    = <<-END
+          A set of identities that will be able to create objects inside the bucket.
+        END
+        readme_example = <<-END
+          lifecycle_rules = [{
+            action = {
+              type          = "SetStorageClass"
+              storage_class = "NEARLINE"
+            }
+            condition = {
+              age                   = 60
+              created_before        = "2018-08-20"
+              with_state            = "LIVE"
+              matches_storage_class = ["REGIONAL"]
+              num_newer_versions    = 10
+            }
+          }]
+        END
+
+        attribute "action" {
+          required    = true
+          type        = list(action)
+          description = <<-END
+            The Lifecycle Rule's action configuration.
+
+            Each `action` object can have the following fields:
```

Comment:
this can be removed.. please check other modules too.. please also alway review rendered result to see if it makes sense ;) thank you ;)